### PR TITLE
Use os.open function for reading files.

### PR DESCRIPTION
**Pull request by Tatsinnit**
This PR kind of closes all the discussion where it was recommended not to use `RunCommandOnHost` for things like `cat` for file because that can be easily done at via `os.Open` which open files. Much better and reduces over-use of `RunCommandonHost`.

* Test with this image: `docker.io/tatsat/redundant-runcommand-use` 

Thanks.

### CODE REVIEW

File: pkg/collector/dns_collector.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ func (collector *DNSCollector) GetName() string {
 func (collector *DNSCollector) Collect() error {
 	result := make(map[string]string)
 
-	output, err := utils.RunCommandOnHost("cat", "/etc/resolv.conf")
+	output, err := utils.ReadFileContent("/etc/resolv.conf")
```
Comment (arnaud-tincelin) [inline pkg/collector/dns_collector.go:27]: The previous command was running on the host, this one will run on the container. Is it working as expected or should we have a mount to do that?

File: pkg/collector/nodelogs_collector.go [hunk 2]
```diff
@@ -29,13 +29,12 @@ func (collector *NodeLogsCollector) Collect() error {
 
 	for _, nodeLog := range nodeLogs {
 
-		output, err := utils.RunCommandOnHost("cat", nodeLog)
+		output, err := utils.ReadFileContent(nodeLog)
```
Comment (arnaud-tincelin) [inline pkg/collector/nodelogs_collector.go:32]: same here

**Review by arnaud-tincelin — CHANGES_REQUESTED** [review]

File: pkg/collector/dns_collector.go (hunk 1 above)
Comment (Tatsinnit) [reply pkg/collector/dns_collector.go:27]: Yeah, so they are already there: `docker.io/tatsat/redundant-runcommand-use` image shows the correct log.

**Comment by Tatsinnit** [comment]
💡 FYI: closing this in favour for this PR: https://github.com/Azure/aks-periscope/pull/109 - it makes sense to do this individually at the each collector level when adding tests. Which will lead to better progress. Thanks heaps.

**Pull request closed without merging**